std/assembly/internal/typedarray.ts

```diff
@@ -231,3 +232,52 @@ export function EVERY<TArray extends TypedArray<T>, T>(
   }
   return true;
 }
+
+@inline
+export function FOR_EACH<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => void,
+): void {
+  var buffer = array.buffer;
+  var length = array.length;
+  var byteOffset = array.byteOffset;
+
+  for (let i = 0; i < length; i++) {
+    callbackfn(LOAD<T>(buffer, i, byteOffset), i, array);
+  }
+}
+
+@inline
+export function FILTER<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => bool,
+): TArray {
+  var backingBuffer = allocateUnsafe(sizeof<T>());
+  var count = 0;
+  var buffer = array.buffer;
+  var length = array.length;
+  var byteOffset = array.byteOffset;
+  var backByteLength = backingBuffer.byteLength;
+  var value: T;
+  for (let i = 0; i < length; i++) {
+    value = LOAD<T>(buffer, i, byteOffset);
+    if (callbackfn(value, i, array)) {
+      backByteLength = backingBuffer.byteLength;
+      if (backByteLength < (count + 1) << alignof<T>()) {
+        backingBuffer = reallocateUnsafe(backingBuffer, backByteLength << 1, false);
+      }
+      STORE<T, NATIVE<T>>(backingBuffer, count, <NATIVE<T>>value);
+      ++count;
+    }
+  }
+
+  // reallocateUnsafe calls this to reduce ArrayBuffer size
+  store<i32>(changetype<usize>(backingBuffer) + offsetof<ArrayBuffer>("byteLength"), count << alignof<T>());
+
+  // instantiate the result
+  var result = instantiate<TArray>(0);
+  store<ArrayBuffer>(changetype<usize>(result) + offsetof<TArray>("buffer"), backingBuffer);
```

Comment:
Could you try move `offsetof<TArray>("buffer")` and `offsetof<TArray>("byteLength")` which actually compile time consts to last `offset` param like:
```ts
store<ArrayBuffer>(changetype<usize>(result), backingBuffer, offsetof<TArray>("buffer"));
```

Reply:
Sounds great! I had no idea this was possible.